presto-hive/src/main/java/com/facebook/presto/hive/clustering/Distribution.java

```diff
@@ -0,0 +1,41 @@
+package com.facebook.presto.hive.clustering;
+
+import com.facebook.presto.hive.HiveType;
+import javafx.util.Pair;
+import org.apache.avro.generic.GenericData;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// Assume: We have 100 values to represent 100 percentiles
+// for the column for now.
+public final class Distribution
+{
+    private List<Pair<Integer, Object>> distribution;
```

Comment:
if `Integer` is to indicate the index, we don't have to use it. Maybe just an object array would be self-explanatory.

The closed data structure to this would be `EquatableValueSet`. An interesting way to store the value is by reusing `Block` and `Type`. We can chat in person about the details.